processing/src/main/java/io/druid/segment/incremental/IncrementalIndex.java

```diff
@@ -578,6 +579,43 @@ public Integer getDimensionIndex(String dimension)
     return dimensionOrder.get(dimension);
   }
 
+  public LinkedHashMap<String, Integer> getDimensionOrder()
+  {
+    return dimensionOrder;
+  }
+
+  /*
+   * Currently called to initialize IncrementalIndex dimension order during index creation
+   * Index dimension ordering could be changed to initalize from DimensionsSpec after resolution of
+   * https://github.com/druid-io/druid/issues/2011
+   */
+  public void loadDimensionOrder(LinkedHashMap<String, Integer> oldDimensionOrder)
```

Comment:
This method and `loadDimensionIterable` are doing basically much the same thing. How about removing this and just always using loadDimensionIterable?

Like, `loadDimensionIterable(aDimensionOrder.keySet())`